config/config.go

```diff
@@ -283,12 +274,7 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
 
 			discovery := discovery.NewRoutingDiscovery(crouter)
 
-			if hop {
-				// advertise ourselves
-				relay.Advertise(ctx, discovery)
-			} else {
-				_ = relay.NewAutoRelay(ctx, h, discovery, router, cfg.StaticRelays)
-			}
+			_ = relay.NewAutoRelay(ctx, h, discovery, router, cfg.StaticRelays)
```

Comment:
I like this. This will make it a lot easier to implement a `Close` method here, and not rely on the `ctx` any more.